chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -59,18 +80,84 @@ impl RpcError {
     }
     /// Create an invalid request error.
     pub fn invalid_request() -> Self {
-        RpcError::new(-32_600, "Invalid request".to_owned(), None)
+        RpcError {
+            code: -32_600,
+            message: "Invalid request".to_owned(),
+            data: None,
+            error_struct: Some(RpcErrorKind::ValidationError(ValidationErrorKind::InvalidRequest)),
+        }
     }
     /// Create a parse error.
     pub fn parse_error(e: String) -> Self {
-        RpcError::new(-32_700, "Parse error".to_owned(), Some(Value::String(e)))
+        RpcError {
+            code: -32_700,
+            message: "Parse error".to_owned(),
+            data: Some(Value::String(e.clone())),
+            error_struct: Some(RpcErrorKind::ValidationError(ValidationErrorKind::ParseError {
+                error_message: e,
+            })),
+        }
     }
     pub fn serialization_error(e: String) -> Self {
-        RpcError::new(-32_000, "Server error".to_owned(), Some(Value::String(e)))
+        RpcError {
+            code: -32_000,
+            message: "Server error".to_owned(),
+            data: Some(Value::String(e.clone())),
+            error_struct: Some(RpcErrorKind::InternalError(serde_json::json!({
+                "name": "SERIALIZATION_ERROR",
+                "info": serde_json::json!({
+                    "error_message": e
+                })
+            }))),
+        }
+    }
+
+    /// Helper method to define extract INTERNAL_ERROR in separate RpcErrorKind
+    /// Returns HANDLER_ERROR if the error is not internal one
+    pub fn new_internal_or_handler_error(error_data: Option<Value>, error_struct: Value) -> Self {
```

Comment:
I am not a huge fan of such methods, but at least it is explicit about what it does. Let's go ahead with it for now